infraestructure/api-gateway/src/app.controller.ts

```diff
@@ -37,4 +37,14 @@ export class AppController {
   getAuth(): Observable<any> {
     return this.authService.send({ cmd: 'Auth' }, {});
   }
+
+  @Get('engage/exercises')
+  getExercises(): Observable<any> {
+    return this.engageHubService.send({ cmd: 'getExercises' }, {});
+  }
+
+  @Get('engage/exercises/filterByName')
+  filterExercises(@Query('name') name: string): Observable<any> {
+    return this.engageHubService.send({ cmd: 'filterExercisesByName' }, { name });
```

Comment:
You should separate endpoint in different controllers not all in `app.controller.ts`